From e715deb3dd8cdc489b91b488676c9ecf684666a5 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 23 Mar 2014 14:01:31 +0000
Subject: [PATCH 6/8] findIcon: use RemoteFile::FindFile() to search for the
 icons

FindFile() is a lot more efficient than Exists() since it only looks in the
specified storage group.
---
 mythtv/libs/libmythmetadata/musicutils.cpp |   19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/mythtv/libs/libmythmetadata/musicutils.cpp b/mythtv/libs/libmythmetadata/musicutils.cpp
index 3de948a..9a82e49 100644
--- a/mythtv/libs/libmythmetadata/musicutils.cpp
+++ b/mythtv/libs/libmythmetadata/musicutils.cpp
@@ -33,33 +33,38 @@ QString fixFilename(const QString &filename)
 static QMap<QString, QString> iconMap;
 QString findIcon(const QString &type, const QString &name, bool ignoreCache)
 {
+    LOG(VB_FILE, LOG_INFO, QString("findicon: looking for type: %1, name: %2").arg(type).arg(name));
+
     if (!ignoreCache)
     {
         QMap<QString, QString>::iterator i = iconMap.find(type + name);
         if (i != iconMap.end())
+        {
+            LOG(VB_FILE, LOG_INFO, QString("findicon: found in cache %1").arg(i.value()));
             return i.value();
+        }
     }
 
     QString cleanName = fixFilename(name);
     QString file = QString("Icons/%1/%2").arg(type).arg(cleanName);
     QStringList imageExtensions = QStringList() << ".jpg" << ".jpeg" << ".png" << ".gif";
+    QString filename;
 
     // TODO also look on any slave BEs?
-    QString filename = gCoreContext->GenMythURL(gCoreContext->GetMasterHostName(),
-                                                0, file, "MusicArt");
-
     for (int x = 0; x < imageExtensions.count(); x++)
     {
-        if (RemoteFile::Exists(filename + imageExtensions[x]))
+        filename = RemoteFile::FindFile(file + imageExtensions[x], gCoreContext->GetMasterHostName(), "MusicArt");
+        if (!filename.isEmpty())
         {
-            iconMap.insert(type + name, filename + imageExtensions[x]);
-            return filename + imageExtensions[x];
+            LOG(VB_FILE, LOG_INFO, QString("findicon: found at %1").arg(filename));
+            iconMap.insert(type + name, filename);
+            return filename;
         }
     }
 
     iconMap.insert(type + name, QString());
 
-    LOG(VB_FILE, LOG_INFO, QString("findicon: not found for type: %1, name: %2").arg(type).arg(name));
+    LOG(VB_FILE, LOG_INFO, QString("findicon: not found type: %1, name: %2").arg(type).arg(name));
 
     return QString();
 }
-- 
1.7.10.2

